src/test/java/com/exasol/spark/s3/S3CleanupIT.java

```diff
@@ -213,10 +210,30 @@ void testSinkJobAlwaysFailsJobEndCleanup() {
      * This class keeps count of failures among multiple Spark runner threads.
```

Comment:
We maybe should extend on this explanation, why we need a class variable (aka singleton) for this, instead of using an object that is shared between the threads. Aka, objects that go into a dataflow will be serialized and with that copied. They don't stay the same instance. However, the Class instances are shared JVM wide and with that also class variables.

Reply:
Another thing, I am thinking about, are accumulators an option, but I am not sure if you can access the value inside the dataflow

https://spark.apache.org/docs/2.2.0/rdd-programming-guide.html#accumulators

Reply:
Yes accumulators could work, but maybe keeping class is good option to make it more visible for what we try to check. 

I will update the comment, it good idea to make it clear.